src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/DecimalMathFuncs.txt

```diff
@@ -0,0 +1,453 @@
+
+>> Abs( 10000000000000000000000001 )
+10000000000000000000000001
+
+>> Abs( -10000000000000000000000001 )
+10000000000000000000000001
+
+>> Abs( [10000000000000000000000001, 20000000000000000000000002, -10000000000000000000000001, -20000000000000000000000002])
+Table({Value:10000000000000000000000001},{Value:20000000000000000000000002},{Value:10000000000000000000000001},{Value:20000000000000000000000002})
+
+>> Abs( [10000000000000000000000001, 20000000000000000000000002, -10000000000000000000000001, -20000000000000000000000002, Float(1)])
+Table({Value:10000000000000000000000001},{Value:20000000000000000000000002},{Value:10000000000000000000000001},{Value:20000000000000000000000002},{Value:1})
+
+// decimal aware functions with mixed inputs
+
+>> Int( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(100000.12)] )
+Table({Value:1000000000000000000000001},{Value:2000000000000000000000002},{Value:100000})
+
+>> Trunc( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(100000.12)] )
+Table({Value:1000000000000000000000001},{Value:2000000000000000000000002},{Value:100000})
+
+>> Round( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.12})
+
+>> RoundUp( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(300000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:300000.13})
+
+>> RoundDown( [1000000000000000000000001.1, 2000000000000000000000002.2, Float(30000.1234)], 2 )
+Table({Value:1000000000000000000000001.1},{Value:2000000000000000000000002.2},{Value:30000.12})
+
+// these rounding tests are returning the correct results, due to the conversion between decimal and float
+
+>> Int( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Trunc( [Float(100000.12), 1000000000000000000000001.1, 2000000000000000000000002.2] )
+Table({Value:100000},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> Round( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundUp( [Float(300000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:300000.13},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+>> RoundDown( [Float(30000.1234), 1000000000000000000000001.1, 2000000000000000000000002.2], 2 )
+Table({Value:30000.12},{Value:1000000000000000100000000},{Value:2000000000000000200000000})
+
+// non-decimal aware numeric functions
+
+>> Sqrt( 1524157899404237160815376 )         // exact square of 1234567899876, should be done in float and match next test
```

Comment:
>1524157899404237160815376 [](http://example.com/codeflow?start=9&length=25)

you should try 1524157899404237160815377
(result = 1234567899876.000000000000405000000445678200490442)

Reply:
Good call.  I went even futher and used 1524157899404230000000000 which still rounds to the same number in float, but would be quite different in decimal.